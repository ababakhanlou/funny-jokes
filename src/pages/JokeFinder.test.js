import React from "react";
import { shallow } from "enzyme";
import { localStorageMock } from "../testHelpers/mockLocalStorage";

describe("JokeFinder", () => {
  const mockGetJoke = jest.fn(() => "Joke!");
  let wrapper;

  Object.defineProperty(window, "localStorage", { value: localStorageMock });

  jest.mock("../components", () => ({
    Button: () => <button>Button</button>,
    Card: () => <div>Card</div>,
  }));

  jest.mock("../api/joke", () => ({
    __esModule: true,
    default: mockGetJoke,
  }));

  const JokeFinder = require("./JokeFinder").default;

  wrapper = shallow(<JokeFinder />);

  it("should render correctly", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("should call getNewJoke on mount", () => {
    expect(mockGetJoke).toHaveBeenCalled();
  });

  it("should save the joke when the like button is pressed if the joke is not saved", () => {
    const saveJoke = wrapper.find("Button").first().prop("action");
    saveJoke();
    expect(localStorageMock.getItem("jokes")).toBe("Joke!<#>");
    // when clicked a second time, it should not save again
    saveJoke();
    expect(localStorageMock.getItem("jokes")).toBe("Joke!<#>");
  });

  it("should get a new joke when button is pressed", () => {
    const getNewJoke = wrapper.find("Button").at(1).prop("action");
    getNewJoke();
    expect(mockGetJoke).toHaveBeenCalled();
  });
});
