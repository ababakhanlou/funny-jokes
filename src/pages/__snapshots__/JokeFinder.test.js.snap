// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JokeFinder should render correctly 1`] = `
"<div>
  <div>
    <Card joke=\\"Joke!\\" />
  </div>
  <div>
    <Button action={[Function: bound saveJoke]} name=\\"Like <3\\" />
    <Button action={[Function: bound getNewJoke]} name=\\"NEXT>>>>>\\" />
  </div>
</div>"
`;
